includes/utilities.inc

```diff
@@ -211,6 +211,14 @@ function islandora_multi_importer_remote_file_get($url) {
     $parsed_url = parse_url($url);
     $remote_schemes = array('http', 'https', 'ftp', 'ftps', 'smb', 'nfs');
     if (!isset($parsed_url['scheme']) || (isset($parsed_url['scheme']) &&!in_array($parsed_url['scheme'], $remote_schemes))) {
+
+      // If local file, engage any hook_islandora_multi_importer_remote_file_get and return the real path.
+      $urls = array( );
+      $urls = module_invoke_all('islandora_multi_importer_remote_file_get', $url);
+      if (!empty($urls)) {
+        return drupal_realpath($urls[0]);
```

Comment:
Hook order/precedence is determined by module "weight" as indicated in several web posts...  

```
Hooks are called in order of

    Weight, which is defined per module in the database as {system.weight}. Lower weights come earlier in the calling process.
    Alphabetical, by module name.

By default, modules get assigned a weight of 0, so nearly all hooks in the system run in alphabetical order. Some modules will adjust this in their install hooks so they run earlier or later in the module_invoke_all function.
```

In this case, the first successful hook/fetch invokes a 'return' from the function, so in effect the lowest weighted successful hook is the one that takes control.  If no hooks are defined, or none succeed, then the default  'drupal_realpath' return is executed.  

My Digital Grinnell implementation of this hook is identical to the code supplied in the hook_islandora_multi_importer.api.php file.  

Hope that helps.  If not, let me know.

Reply:
A bit more detail...  my hook is implemented in my DG7 custom module, it's where ALL of my hooks exist.  DG7 has the default module 'weight' of zero so it's hook implementations fire based on the alphabetical order of 'dg7' in the list of all enabled modules.  This puts it relatively early in the calling sequence since the letter 'd' proceeds most others, assuming the others don't explicitly specify a lower module weight of their own.  

However, all of this is moot in my case since DG7 is the only module that implements hook_islandora_multi_importer_remote_file_get.

Reply:
And I found a problem with my hook implementation...working that out now.  In the meantime I've added messages like the following so that folks know what the hook has done...

```
IMI hook dg7_islandora_multi_importer_remote_file_get was invoked and found file 'public://imi_files//Douglas_Peterson_50/Doug_Peterson_50.MP3' for ingest.
IMI hook dg7_islandora_multi_importer_remote_file_get was invoked and found file 'public://imi_files//Douglas_Peterson_50/IOH-Peterson.xml' for ingest.
```